Allow Peer Endpoint to check Interface for VRF validation

jmcgill298 opened the pull request:
<!--
    Thank you for your interest in contributing to BGP Models! Please note
    that our contribution policy recommends that a feature request or bug
    report be opened for approval prior to filing a pull request. This
    helps avoid wasting time and effort on something that we might not
    be able to accept.

    Please indicate the relevant feature request or bug report below.
-->

# Closes: #<ISSUE NUMBER GOES HERE>

## What's Changed

<!--
    Please include:
    - A summary of the proposed changes
    - A sectioned breakdown for larger features under ## subheadings
    - Screenshots, example payloads where relevant:
      - Before/After for bugfixes
      - Using a new feature
-->

Updated PeerEndpoint to validate PeerGroup VRF against Source Interface VRF in addition to Source IP validation. This allows Peering definitions to be used by implementations that associate VRFs with Interfaces and not Prefixes

## To Do

<!--
    Please feel free to update todos to keep track of your own notes for WIP PRs.
-->
- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/core/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

jmcgill298 committed:
Allow Peer Endpoint to check Interface for VRF validation

glennmatthews commented on nautobot_bgp_models/models.py:
```diff
@@ -627,10 +627,17 @@ def clean(self):
+                        f"VRF mismatch between {local_ip_value} (VRF {local_ip_value.parent.vrfs.all().first()}) "
+                        f"and peer-group {self.peer_group.name} (VRF {self.peer_group.vrf})"
+                    )
+                if not source_interface and source_interface.vrf != self.peer_group.vrf:
```
Should this be:

```suggestion
                elif source_interface.vrf != self.peer_group.vrf:
```

glennmatthews reviewed (commented):
1) Can you add a unit test or two exercising this logic?
2) Is a similar fix needed in `develop`?

jmcgill298 committed:
Update nautobot_bgp_models/models.py

Co-authored-by: Glenn Matthews <glenn.matthews@networktocode.com>

jmcgill298 commented:
> 1. Can you add a unit test or two exercising this logic?
> 2. Is a similar fix needed in `develop`?

1) I looked for an existing test case, but could not find anything. Where should a test go?

2) I'm sure PR's will need to be made to multiple branches, just picked ltm-2.4 as a first branch

glennmatthews commented:
I'd expect a test in `test_models.py`, if we don't have such already please feel free to add it. It can inherit from `nautobot.apps.testing.ModelTestCases.BaseModelTestCase`.

jmcgill298 committed:
Add test cases

jmcgill298 commented:
> I'd expect a test in `test_models.py`, if we don't have such already please feel free to add it. It can inherit from `nautobot.apps.testing.ModelTestCases.BaseModelTestCase`.

@glennmatthews I removed the two test cases that were commented out and added a few cases to test the different scenarios/branches for this section of code

jmcgill298 committed:
Fix pylint errors

glennmatthews commented:
Reminder that this will need to be forward-ported to `develop` as well. :-)
